jsonnet/main.jsonnet

```
@@ -1,12 +1,23 @@
 local etcdMixin = (import 'github.com/etcd-io/etcd/contrib/mixin/mixin.libsonnet');
 local openshiftRules = (import 'custom.libsonnet');
-local a = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
-local r = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
-local o = if std.objectHasAll(openshiftRules, 'prometheusRules') then openshiftRules.prometheusRules.groups else [];
+
+local alertingRules = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
+local promRules = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
+local openshiftAlerts = if std.objectHasAll(openshiftRules, 'prometheusRules') then openshiftRules.prometheusRules.groups else [];
```

nit: I think you can put all those under one variable and simplify further filtering. 
```suggestion
local alertingRules = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
local promRules = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
local rules = alertingRules + promRules + openshiftRules.prometheusRules.groups
```

Checking for group inclusion is necessary when you are importing code from another repository. In the case of the same repository (example `openshiftRules.prometheusRules.groups`) it is not so important as you are in full control.

---

Thanks! We can't do this exactly as we need those separate and not in the same group, but sort of went with similar pattern.

---

I think I am missing why those need to be separated as all filters check first if entry is a recording rule or an alert :thinking:

---

The openshift rules we do not want to filter out, only exclude alerts from the upstream mixin.

---

Or did you mean to add promRules+alertingRules together?

---

All good, I need to re-learn how to read though :joy: